### PR TITLE
ZEPPELIN-3737. Wrap JobManager page related stuff into class JobManagerService

### PR DESCRIPTION
**Pull request by zjffdu**
### What is this PR for?

This is refactoring PR which move all JobManager page related stuff into class JobManagerService.

### What type of PR is it?
[Refactoring]

### Todos
* [ ] - Task

### What is the Jira issue?
* https://issues.apache.org/jira/browse/ZEPPELIN-3737

### How should this be tested?
* CI pass

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java [hunk 1]
```diff
@@ -2041,9 +2063,10 @@ public NotebookInformationListener(NotebookServer notebookServer) {
     @Override
     public void onParagraphRemove(Paragraph p) {
       try {
-        notebookServer.broadcastUpdateNoteJobInfo(System.currentTimeMillis() - 5000);
+        getJobManagerService().getNoteJobInfoByUnixTime(System.currentTimeMillis() - 5000, null,
+            new JobManagerServiceCallback());
       } catch (IOException ioe) {
-        LOG.error("can not broadcast for job manager {}", ioe.getMessage());
+        LOG.warn("can not broadcast for job manager {}", ioe.getMessage());
```
Comment (felixcheung) [inline zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java]: LOG.warn("can not broadcast for job manager {}", ioe.getMessage(), ioe);

File: zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java [hunk 2]
```diff
@@ -2052,71 +2075,62 @@ public void onNoteRemove(Note note) {
       try {
         notebookServer.broadcastUpdateNoteJobInfo(System.currentTimeMillis() - 5000);
       } catch (IOException ioe) {
-        LOG.error("can not broadcast for job manager {}", ioe.getMessage());
+        LOG.warn("can not broadcast for job manager {}", ioe.getMessage());
```
Comment (felixcheung) [inline zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java]: LOG.warn("can not broadcast for job manager {}", ioe.getMessage(), ioe);

File: zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java [hunk 3]
```diff
@@ -2052,71 +2075,62 @@ public void onNoteRemove(Note note) {
...
-      response.put("lastResponseUnixTime", System.currentTimeMillis());
-      response.put("jobs", notesInfo);
-
-      notebookServer.broadcast(JobManagerService.JOB_MANAGER_PAGE.getKey(),
-          new Message(OP.LIST_UPDATE_NOTE_JOBS).put("noteRunningJobs", response));
+      try {
+        getJobManagerService().removeNoteJobInfo(note.getId(), null,
+            new JobManagerServiceCallback());
+      } catch (IOException ioe) {
+        LOG.warn("can not broadcast for job manager {}", ioe.getMessage());
```
Comment (felixcheung) [inline zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java]: LOG.warn("can not broadcast for job manager {}", ioe.getMessage(), ioe);

Comment (felixcheung) [reply zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java]: ditto to all other cases

File: zeppelin-server/src/main/java/org/apache/zeppelin/service/JobManagerService.java [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+    private List<ParagraphJobInfo> paragraphs;
+
+    public NoteJobInfo(Note note) {
+      boolean isNoteRunning = false;
+      long lastRunningUnixTime = 0;
+      this.noteId = note.getId();
+      this.noteName = note.getName();
+      // set note type ( cron or normal )
+      if (note.getConfig().containsKey("cron") && !note.getConfig()
+          .get("cron").equals("")) {
```
Comment (felixcheung) [inline zeppelin-server/src/main/java/org/apache/zeppelin/service/JobManagerService.java]: consider adding a helper perhaps? should it check for get("cron") that is not whitespace only?

File: zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java (hunk 1 above)
Comment (zjffdu) [reply zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java]: fixed

**Comment by zjffdu** [comment]
Will merge if no more comments

**Pull request closed without merging**